### PR TITLE
feature/seatid-for-player

### PR DESCRIPTION
**Pull request by danielsuter**
Added seatId for player objects.

### CODE REVIEW

File: server/session/sessionHandler.js [hunk 1]
```diff
@@ -86,8 +86,9 @@ const SessionHandler = {
 
         return clientApi.requestPlayerName(ws).then((playerName) => {
             return clientApi.requestSessionChoice(ws, this.getAvailableSessionNames()).then((sessionChoiceResponse) => {
-                let session = createOrJoinSession(this.sessions, sessionChoiceResponse);
+                let session = createAndReturnSession(this.sessions, sessionChoiceResponse);
```
Comment (dstucki) [inline server/session/sessionHandler.js]: if not reassigning, please use const

File: server/session/singleGameSession.js [hunk 2]
```diff
@@ -73,12 +75,23 @@ function createPlayer(session, webSocket, playerName, chosenTeamIndex) {
     });
 }
 
+// from http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
```
Comment (dstucki) [inline server/session/singleGameSession.js]: use npm install uuid instead?

File: test/client/game/gameStoreTest.js [hunk 3]
```diff
@@ -22,6 +22,11 @@ describe('GameStore', () => {
         GameStore.handlePayload.restore();
     });
 
+    const filterEmptyPlayer = function(gamestoreState) {
+        return gamestoreState.players.filter(player => player.name != 'Waiting for player...');
```
Comment (dstucki) [inline test/client/game/gameStoreTest.js]: !==

File: test/server/session/singleGameSessionTest.js [hunk 4]
```diff
@@ -62,8 +62,8 @@ describe('Session', function () {
         }
 
         function expectPlayerInTeam(playerIndex, playerName, teamIndex) {
-            expect(session.players[playerIndex].team).to.equal(session.teams[teamIndex]);
-            expect(session.players[playerIndex].name).to.equal(playerName);
+            let player = session.players.find(player => player.name === playerName);
```
Comment (dstucki) [inline test/server/session/singleGameSessionTest.js]: use const

File: test/server/session/singleGameSessionTest.js [hunk 5]
```diff
@@ -134,19 +134,25 @@ describe('Session', function () {
         it('should broadcast session joined and save message for later use', () => {
             let sessionPlayer = {
                 id: 0,
+                seatId: 0,
                 name: 'name'
             };
 
-            clientApiMock.expects('broadcastSessionJoined').once().withArgs(session.name, sessionPlayer, [sessionPlayer]);
+            var playerMatcher = sinon.match(function (actualPlayer) {
```
Comment (dstucki) [inline test/server/session/singleGameSessionTest.js]: use let or const

File: test/testDataCreator.js [hunk 6]
```diff
@@ -4,9 +4,9 @@ import * as Deck from '../server/game/deck/deck';
 import * as Player from '../server/game/player/player';
 import * as Team from '../server/game/player/team';
 
-let createDummyPlayer = function (teamName, name, id, clientApiMock) {
+let createDummyPlayer = function (teamName, name, id, seatId, clientApiMock) {
```
Comment (enpayne) [inline test/testDataCreator.js]: just name the function directly, assignment not necessary

File: server/session/singleGameSession.js (hunk 2 above)
Comment (bmillo) [inline server/session/singleGameSession.js]: dependency for 6 line of code?

File: client/js/game/gameStore.js [hunk 7]
```diff
@@ -26,6 +26,16 @@ export const PlayerType = {
     SPECTATOR: 'SPECTATOR'
 };
 
+function emptyPlayer (emptyPlayerId) {
+    return {
+        id: emptyPlayerId + '',
```
Comment (r3dDoX) [inline client/js/game/gameStore.js]: You could use String(emptyPlayerId) for this

File: client/js/game/gameStore.js [hunk 8]
```diff
@@ -26,6 +26,16 @@ export const PlayerType = {
...
+        seatId: emptyPlayerId,
+        name: "Waiting for player..."
```
Comment (r3dDoX) [inline client/js/game/gameStore.js]: Use single quotes

File: server/session/singleGameSession.js (hunk 2 above)
Comment (danielsuter) [inline server/session/singleGameSession.js]: leftpad?

**Pull request merged**